core/setup/kubernetes.go

```diff
@@ -47,47 +52,71 @@ func kubernetesParse(c *Controller) (kubernetes.Kubernetes, error) {
      * API calls may be desirable.
      */ 
 
-	k8s := kubernetes.Kubernetes{
+    k8s := kubernetes.Kubernetes{
         Proxy:      proxy.New([]string{}),
-		Ctx:        context.Background(),
-//		Inflight:   &singleflight.Group{},
+        Ctx:        context.Background(),
+//      Inflight:   &singleflight.Group{},
         APIConn:    nil,
-	}
-	var (
-		endpoints     = []string{defaultK8sEndpoint}
-	)
-	for c.Next() {
-		if c.Val() == "kubernetes" {
-			k8s.Zones = c.RemainingArgs()
-			if len(k8s.Zones) == 0 {
-				k8s.Zones = c.ServerBlockHosts
-			}
-			middleware.Zones(k8s.Zones).FullyQualify()
-			if c.NextBlock() {
-				// TODO(miek): 2 switches?
-				switch c.Val() {
-				case "endpoint":
-					args := c.RemainingArgs()
-					if len(args) == 0 {
-						return kubernetes.Kubernetes{}, c.ArgErr()
-					}
-					endpoints = args
+        NameTemplate: nil,
```

Comment:
You don't really need to specify all these nil values. All elements will get their natural zero value, which for pointers is nil.